simtools/corsika/corsika_output.py

```diff
@@ -976,44 +1036,416 @@ def num_photons_per_event(self):
+                    "file name": "hist_2D_photon_density_distr",
+                    "title": "Photon distribution on the ground",
+                    "x edges": "x position on the ground",
+                    "x edges unit": self.hist_config["hist_position"]["x axis"]["start"].unit,
+                    "y edges": "y position on the ground",
+                    "y edges unit": self.hist_config["hist_position"]["y axis"]["start"].unit,
+                },
+            }
+        return self.__dict_2D_distributions
+
+    def _export_2D_histograms(self, output_dir="./"):
+        """
+        Auxiliary function to export only the 2D histograms.
+
+        Parameters
+        ----------
+        output_dir: str
+            Output directory, where to save the histograms.
+        """
+
+        for property_name, function_dict in self._dict_2D_distributions.items():
+            self._meta_dict["Title"] = function_dict["title"]
+            function = getattr(self, function_dict["function"])
+
+            hist_2D_list, x_edges_list, y_edges_list = function()
+            if function_dict["function"] == "get_2D_photon_density_distr":
+                histogram_value_unit = 1 / (
+                        self._dict_2D_distributions[property_name]["x edges unit"]
+                        * self._dict_2D_distributions[property_name]["y edges unit"]
+                )
+            else:
+                histogram_value_unit = u.dimensionless_unscaled
+
+            hist_2D_list, x_edges_list, y_edges_list = (
+                hist_2D_list * histogram_value_unit,
+                x_edges_list * self._dict_2D_distributions[property_name]["x edges unit"],
+                y_edges_list * self._dict_2D_distributions[property_name]["y edges unit"],
+            )
+
+            for i_histogram, _ in enumerate(x_edges_list):
+                if self.individual_telescopes:
+                    ecsv_file = (
+                        f"{self._dict_2D_distributions[property_name]['file name']}"
+                        f"_tel_index_{self.telescope_indices[i_histogram]}.ecsv"
+                    )
+                else:
+                    ecsv_file = (
+                        f"{self._dict_2D_distributions[property_name]['file name']}_all_tels.ecsv"
+                    )
+
+                table = self.fill_ecsv_table(
+                    hist_2D_list[i_histogram],
+                    x_edges_list[i_histogram],
+                    y_edges_list[i_histogram],
+                    function_dict["x edges"],
+                    function_dict["y edges"],
+                )
+                ecsv_file = Path(output_dir).joinpath(ecsv_file)
+                print(ecsv_file)
+                self._logger.info(f"Exporting histogram to {ecsv_file}.")
+                table.write(ecsv_file, format="ascii.ecsv", overwrite=True)
+
+    def fill_ecsv_table(self, hist, x_edges, y_edges, x_label, y_label):
```

Comment:
Do you think it would be confusing if you change the order such that the `x` values are first and then the `y` values? It would have the advantage that you can then set a default `None` to the `y` values and avoid explicitly giving it in the fill command above. You can also then add a comment to the `except` clause below, explaining that in the case `y` is `None` you will get a `TypeError` (I had to try it to figure that out).

Reply:
It would be confusing because we are dealing with histograms here. In one of the previous PRs, after your comment about the order of the histograms (x or y first), I established the convention to always use the histogram values first (y), which is the numpy way as well.

Reply:
OK, not a big deal, we can keep it as is.